mlrun/package/packager.py

```diff
@@ -0,0 +1,537 @@
+
+        :param obj: The object to pack and log.
+        :param key: The result's key.
+
+        :return: The result dictionary.
+        """
+        return {key: obj}
+
+    @classmethod
+    def unpack_object(
+        cls,
+        data_item: DataItem,
+        pickle_module_name: str = _DEFAULT_PICKLE_MODULE,
+        object_module_name: str = None,
+        python_version: str = None,
+        pickle_module_version: str = None,
+        object_module_version: str = None,
+    ) -> Any:
+        """
+        Unpack the data item's object, unpickle it using the instructions and return.
+
+        Warnings of mismatching python and module versions between the original pickling interpreter and this one may be
+        raised.
+
+        :param data_item:             The data item holding the pkl file.
+        :param pickle_module_name:    Module to use for unpickling the object.
+        :param object_module_name:    The original object's module. Used to verify the current interpreter object module
+                                      version match the pickled object version before unpickling the object.
+        :param python_version:        The python version in which the original object was pickled. Used to verify the
+                                      current interpreter python version match the pickled object version before
+                                      unpickling the object.
+        :param pickle_module_version: The pickle module version. Used to verify the current interpreter module version
+                                      match the one who pickled the object before unpickling it.
+        :param object_module_version: The original object's module version to match to the interpreter's module version.
+
+        :return: The un-pickled python object.
+        """
+        # Get the pkl file to local directory:
+        pickle_path = data_item.local()
+
+        # Unpickle and return:
+        return _Pickler.unpickle(
+            pickle_path=pickle_path,
+            pickle_module_name=pickle_module_name,
+            object_module_name=object_module_name,
+            python_version=python_version,
+            pickle_module_version=pickle_module_version,
+            object_module_version=object_module_version,
+        )
+
+    class _ArgumentsType:
+        """
+        Library class for the arguments type to send for `_validate_method_arguments`. Configurations is the term for
+        the kwargs sent to packing methods and instructions are the term for the kwargs sent in unpacking methods.
+        """
+
+        INSTRUCTIONS = "instructions"
+        CONFIGURATIONS = "configurations"
+
+    @classmethod
+    def _get_supported_artifact_types(cls) -> List[str]:
+        """
+        Get all the supported artifact types on this packager.
+
+        :return: A list of all the supported artifact types.
+        """
+        return [key[len("pack_") :] for key in cls.__dict__ if key.startswith("pack_")]
```

Comment:
for `artifact_type="result"` there is no `unpack`, but that should be the only case and results are not passed as `DataItem`s so this case will never happen.

Reply:
I understand, hence my question - let's say someone wrote a packager and add a `pack_x` but forgot `unpack_x`. This will make `x` a supported artifact type even though the class can't unpack it. Seems to me better to fail early and say you cannot pack the object rather than do some packing and no unpacking and fail in some point in the future where it's harder to figure out what happened.